YatimaStdLib/List.lean

```diff
@@ -39,4 +39,83 @@ instance : Foldable List where
   foldr := List.foldr
   foldl := List.foldl
 
+-- the `B` suffix avoids name conflicts with mathlib
+def eraseDupB [BEq α] : List α → List α
+  | [] => []
+  | x::xs => 
+    let exs := eraseDupB xs
+    if exs.contains x then exs else x::exs
+
+def splitAtP [BEq α] (p : α → Bool) (l : List α) : List α × List α :=
+  match l.dropWhile p with 
+  | [] => unreachable!
+  | a::as => ⟨l.takeWhile p ++ [a], as⟩
+
+partial def mergeM [Monad μ] (cmp: α → α → μ Ordering) : List α → List α → μ (List α)
+  | as@(a::as'), bs@(b::bs') => do
+    if (← cmp a b) == Ordering.gt
+    then List.cons b <$> mergeM cmp as bs'
+    else List.cons a <$> mergeM cmp as' bs
+  | [], bs => return bs
```

Comment:
`return` and `pure` are the same, I think return is just colored as a keyword because of its association as one in every other language. 

There aren't any coercions being used here, I think. 

It's  `partial` because we would need to prove termination otherwise; `partial` doesn't mark a partial function

Reply:
O-o-o-o, thank you for explanation about the meaning of `partial`!

I don't know, I sometimes had errors when I tried `return`, but no errors with `pure`. :)